src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,187 @@
+        return this.isExit;
+    }
+
+    /**
+     * Parses the user input and runs the corresponding command.
+     *
+     * @param readIn A string containing the user input.
+     * @return A TaskList containing the updated Tasks after command is parsed.
+     * @throws DukeException A Duke-specific exception that occurs when user input is parsed.
+     */
+    public TaskList parse(String readIn) throws DukeException {
+        if (readIn.equals("bye")) {
+            Ui.printReply("Bye. Hope to see you again soon!");
+            this.isExit = true;
+        } else if (readIn.equals("list")) {
+            list(this.tasks);
+        } else {
+            String[] commandArguments = readIn.split(" ", 2);
+            String command = commandArguments[0];
+            String arguments = "";
+            if (commandArguments.length == 2) {
+                arguments = commandArguments[1];
+            }
+
+            switch (command) {
+            case "done": {
+                if (commandArguments.length < 2) {
+                    throw new DukeException("☹ OOPS!!! The index of '" + command + "' cannot be empty.");
+                }
+                this.toRewriteData = true;
+                int counter = Integer.parseInt(arguments);
+                done(counter);
+                break;
+            }
+            case "deadline": {
+                if (commandArguments.length < 2) {
+                    throw new DukeException("☹ OOPS!!! The description of '" + command + "' cannot be empty.");
+                }
+                this.toRewriteData = true;
+                String[] splitTask = arguments.split(" /by ");
+                if (splitTask.length < 2) {
+                    throw new DukeException("Please indicate a deadline using '/by'.");
+                }
+                String description = splitTask[0];
+                String byString = splitTask[1];
+                try {
+                    LocalDateTime by = LocalDateTime.parse(byString, FORMATTER);
+                    this.tasks.addTask(new Deadline(description, by));
+                } catch (DateTimeParseException e) {
+                    Ui.printReply("Datetime should be in YYYY-MM-DD hr:min (24h clock) format.");
+                }
+                break;
+            }
+            case "event": {
+                if (commandArguments.length < 2) {
+                    throw new DukeException("☹ OOPS!!! The description of '" + command + "' cannot be empty.");
+                }
+                this.toRewriteData = true;
+                String[] splitTask = arguments.split(" /at ");
+                if (splitTask.length < 2) {
+                    throw new DukeException("Please indicate the event time frame using '/at'.");
+                }
+                String description = splitTask[0];
+                String atString = splitTask[1];
+                try {
+                    LocalDateTime at = LocalDateTime.parse(atString, FORMATTER);
```

Comment:
can reconsider naming of var "at", perhaps better would be "eventTime" / "dateTime" / "time"